src/peer-store/README.md

```diff
@@ -10,7 +10,9 @@ Several libp2p subsystems will perform operations, which will gather relevant in
 
 In a libp2p node's life, it will discover peers through its discovery protocols. In a typical discovery protocol, addresses of the peer are discovered along with its peer id. Once this happens, the PeerStore should collect this information for future (or immediate) usage by other subsystems. When the information is stored, the PeerStore should inform interested parties of the peer discovered (`peer` event).
 
-Taking into account a different scenario, a peer might perform/receive a dial request to/from a unkwown peer. In such a scenario, the PeerStore must store the peer's multiaddr once a connection is established. 
+Taking into account a different scenario, a peer might perform/receive a dial request to/from a unkwown peer. In such a scenario, the PeerStore must store the peer's multiaddr once a connection is established.
+
+When a connection is being upgraded, more precisely after its encryption, or even in a discovery protocol, a libp2p node can get to know other parties public keys. In this scenario, libp2p will add the peer's public key to its `KeyBook`.
```

Comment:
@jacobheun I could not do this. 

I think that it makes sense, but afaik noise uses static public keys, which I would not expect to have here. In addition, we will need to provide libp2p to the crypto module for adding the key to the keyBook or change the crypto interface to also require the public key, so that we can update this in the `upgrader.js`.

We should probably give some thought on this and follow up with a good solution.

Reply:
Noise has its own keycache and I dont think we need to worry about persistence of the crypto transport keys, just the libp2p id keys. I think this section is achieved by updating the peer after we've established a connection, because the crypto handshake will result in libp2p public key exchange and verification.

Reply:
For the inbound connection, we do have the public key after the connection being established. The same is not true for the outbound connection. Both for `libp2p-secio` and `libp2p-noise`

I went deeper on secio to figure out why this is happening and I found this: https://github.com/libp2p/js-libp2p-secio/blob/master/src/handshake/crypto.js#L66-L73

When an inbound connection, we go to the else statement while in an outbound connection we go to the if side of things. In the `if` part we basically do not add the public key. If I just add `state.id.remote = remoteId` inside the if and after the validation condition, it works. Is this the expected?

Reply:
In noise, I could not follow the flow so easily yet

Reply:
Note: this is fixed in secio now, I'll look into it in noise as part of my work there.

Reply:
what do you recommend here? change the test configuration to use secio for now and create an issue to track this?

Reply:
Leave it for now, I'll work on getting a patch for noise to correct this. We can add a TODO item to the peer store epic to track getting this in.